packages/driver/src/cy/mouse.js

```diff
@@ -483,9 +483,19 @@ const create = (state, keyboard, focused, Cypress) => {
         }
       }
 
-      if (shouldMoveCursorToEndAfterMousedown(el)) {
-        debug('moveSelectionToEnd due to click', el)
-        $selection.moveSelectionToEnd(el, { onlyIfEmptySelection: true })
+      if (shouldMoveCursorAfterMousedown(el)) {
+        debug('moveSelection to', caretPosition)
+        debug('moveSelection due to click', el)
+
+        if (caretPosition === 'start') {
+          $selection.moveSelectionToStart(el, { onlyIfEmptySelection: true })
+        } else if (caretPosition === 'end') {
+          $selection.moveSelectionToEnd(el, { onlyIfEmptySelection: true })
+        } else if (caretPosition === 'point') {
+          const { x, y } = getMouseCoords(state)
+
+          $selection.moveSelectionToPoint(state('window'), x, y)
+        }
```

Comment:
similarly needs `onlyIfEmptySelection: true` to fulfill the browser default action if selection is changed during the prev event

Reply:
It seems that it's impossible. To use `onlyIfEmptySelection`, We need element. But in this case, we don't use element. We just use `clientX` and `clientY`.